### PR TITLE
EditLabStatus command

### PR DESCRIPTION
**Pull request by channne**
Add functionality for EditLabStatusCommand
Some changes are copied from Ryan's PR (used some functions he made to minimize later merge conflicts)

### CODE REVIEW

File: src/main/java/seedu/address/model/person/lab/LabList.java [hunk 1]
```diff
@@ -37,6 +37,34 @@ public boolean contains(Lab toCheck) {
...
+
+        throw new LabNotFoundException(labToGet.labNumber);
+    }
+
+    public Lab getLabByLabNumber(int labNumberToGet) throws LabNotFoundException {
+        requireNonNull(labNumberToGet);
+
+        for (Lab l : internalList) {
+            if (l.labNumber == labNumberToGet) {
+                return l.createCopy();
```
Comment (geetengtan) [inline src/main/java/seedu/address/model/person/lab/LabList.java:61]: I might be mistaken here, but <code>.createCcopy()</code> seems redundant here?

**Review by geetengtan — COMMENTED** [review]
Very small issue, LGTM

**Review by bryans17 — APPROVED** [review]
Very clean code! No obvious issues. Just some questions. Otherwise LGTM 👍🏻.

Thanks for changing all the `person` to `student` and `addressbook` to `TAddressbook`. Really appreciate it.

File: src/main/java/seedu/address/logic/commands/EditLabStatusCommand.java [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+    public static final String MESSAGE_INVALID_STATUS_CHANGE = "New lab status provided is identical to previous";
+    // add this to commons.core.Messages if it also applies to other commands
+    public static final String MESSAGE_INVALID_LAB_NUMBER = "The lab number provided is invalid";
+
+    private final Index index;
+    private final int labNumber;
+    private final LabStatus newStatus;
+
+    /**
+     *
```
[the remark this answers is not in the document]
Comment (channne) [reply src/main/java/seedu/address/logic/commands/EditLabStatusCommand.java:47]: Nice catch, didn't notice that

File: src/main/java/seedu/address/model/util/SampleDataUtil.java [hunk 3]
```diff
@@ -27,7 +27,7 @@
                 new GithubUsername("alexyeoh"), new Telegram("alex_yeoh"), new StudentId("A0123456B")),
             new Person(new Name("Bernice Yu"), new Email("berniceyu@example.com"), getTagSet("colleagues", "friends"),
                 new GithubUsername("berniceyu"), new Telegram("bernice_yu"), new StudentId("A1234567C")),
-            new Person(new Name("Charlotte Oliveiro"), new Email("charlotte@example.com"), getTagSet("neighbours"),
+            new Person(new Name("Charlotte Oliveiro"), new Email("charlotte@example.com"), getTagSet("neighbors"),
```
[the remark this answers is not in the document]
Comment (channne) [reply src/main/java/seedu/address/model/util/SampleDataUtil.java:30]: 😆 not sure how strict they are about the using American spelling thing

**Pull request merged**